base/src/main/java/org/aya/tyck/unify/DefEq.java

```diff
@@ -303,19 +304,47 @@ private boolean doCompareTyped(@NotNull Term type, @NotNull Term lhs, @NotNull T
       });
       // TODO: path lambda conversion
       case FormTerm.Path path -> throw new UnsupportedOperationException("TODO");
-      case FormTerm.PartTy ty && lhs instanceof IntroTerm.PartEl lel && rhs instanceof IntroTerm.PartEl rel
-        && lel.partial() instanceof Partial.Const<Term> ll
-        && rel.partial() instanceof Partial.Const<Term> rr -> doCompareTyped(ty.type(), ll.u(), rr.u(), lr, rl);
-      case FormTerm.PartTy ty && lhs instanceof IntroTerm.PartEl lel && rhs instanceof IntroTerm.PartEl rel
-        && !(lel.partial() instanceof Partial.Const<Term>)
-        && !(rel.partial() instanceof Partial.Const<Term>) -> CofThy.conv(ty.restr(), new Subst(),
-        subst -> doCompareTyped(ty.subst(subst), lhs.subst(subst), rhs.subst(subst), lr, rl));
+      case FormTerm.PartTy ty && lhs instanceof IntroTerm.PartEl lel && rhs instanceof IntroTerm.PartEl rel ->
+        comparePartial(lel, rel, ty, lr, rl);
       case FormTerm.PartTy ty -> false;
     };
     traceExit();
     return ret;
   }
 
+  private boolean comparePartial(
+    @NotNull IntroTerm.PartEl lhs, @NotNull IntroTerm.PartEl rhs,
+    @Nullable FormTerm.PartTy type, Sub lr, Sub rl
+  ) {
+    return switch (lhs.partial()) {
+      case Partial.Const<Term> ll && rhs.partial() instanceof Partial.Const<Term> rr ->
+        compare(ll.u(), rr.u(), lr, rl, type == null ? null : type.type());
+      case Partial.Split<Term> ll && rhs.partial() instanceof Partial.Split<Term> rr ->
+        CofThy.conv(type == null ? ll.restr() : type.restr(), new Subst(),
+          subst -> compare(lhs.subst(subst), rhs.subst(subst), lr, rl, type == null ? null : type.subst(subst)));
+      default -> false;
+    };
+  }
+
+  private boolean compareCube(@NotNull Cube<Term> lhs, @NotNull Cube<Term> rhs, Sub lr, Sub rl) {
+    lhs.params().zipView(rhs.params()).forEach(x -> {
+      lr.map.put(x._1, new RefTerm(x._2));
+      rl.map.put(x._2, new RefTerm(x._1));
+    });
+    // TODO: let CofThy.propExt uses lr and rl?
```

Comment:
Can we just capture these variables in the lambda in propExt?

Reply:
What do you mean by "capture" (sorry I didn't get it)

Reply:
We can address this later :)